<a href="https://colab.research.google.com/github/yoonputer/test_visuallization/blob/master/autompg_pandas_info.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [2]:
!pwd

/content


In [3]:
!ls -l ./auto-mpg.csv

-rw-r--r-- 1 root root 21913 Jul  5 01:19 ./auto-mpg.csv


In [6]:
df = pd.read_csv('./auto-mpg.csv',header= None)

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 398 entries, 0 to 397
Data columns (total 9 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   0       398 non-null    float64
 1   1       398 non-null    int64  
 2   2       398 non-null    float64
 3   3       398 non-null    object 
 4   4       398 non-null    float64
 5   5       398 non-null    float64
 6   6       398 non-null    int64  
 7   7       398 non-null    int64  
 8   8       398 non-null    object 
dtypes: float64(4), int64(3), object(2)
memory usage: 28.1+ KB


RangeIndex: 398 entries, 0 to 397 : 행 의 사이즈  
Data columns (total 9 columns): 열 의 사이즈  
dtypes: float64(4), int64(3), object(2) : 데이터셋

In [8]:
df.describe()

0           1           2  ...           5           6           7
count  398.000000  398.000000  398.000000  ...  398.000000  398.000000  398.000000
mean    23.514573    5.454774  193.425879  ...   15.568090   76.010050    1.572864
std      7.815984    1.701004  104.269838  ...    2.757689    3.697627    0.802055
min      9.000000    3.000000   68.000000  ...    8.000000   70.000000    1.000000
25%     17.500000    4.000000  104.250000  ...   13.825000   73.000000    1.000000
50%     23.000000    4.000000  148.500000  ...   15.500000   76.000000    1.000000
75%     29.000000    8.000000  262.000000  ...   17.175000   79.000000    2.000000
max     46.600000    8.000000  455.000000  ...   24.800000   82.000000    3.000000

[8 rows x 7 columns]

In [10]:
df[3].describe()

count       398
unique       94
top       150.0
freq         22
Name: 3, dtype: object

In [11]:
df[8].describe()

count            398
unique           305
top       ford pinto
freq               6
Name: 8, dtype: object

In [18]:
df[0].mean(), df[0].std(), df[5].count(), df[5].min(), df[5].max()

(23.514572864321615, 7.815984312565782, 398, 8.0, 24.8)